app/src/main/java/org/mozilla/focus/persistence/TabModel.java

```diff
@@ -0,0 +1,148 @@
+package org.mozilla.focus.persistence;
+
+import android.arch.persistence.room.ColumnInfo;
+import android.arch.persistence.room.Entity;
+import android.arch.persistence.room.Ignore;
+import android.arch.persistence.room.PrimaryKey;
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+
+@Entity(tableName = "tabs")
+public class TabModel {
+
+    @Ignore
+    public TabModel(String id, String parentId) {
+        this(id, parentId, "", "", "", "");
+    }
+
+    @Ignore
+    public TabModel(String id, String parentId, String title, String url) {
+        this(id, parentId, title, url, "", "");
+    }
+
+    public TabModel(String id, String parentId, String title, String url, String thumbnailUri, String webViewStateUri) {
+        this.id = id;
+        this.parentId = parentId;
+        this.title = title;
+        this.url = url;
+        this.thumbnailUri = thumbnailUri;
+        this.webViewStateUri = webViewStateUri;
+    }
+
+    @PrimaryKey
+    @NonNull
+    @ColumnInfo(name = "tab_id")
+    private String id;
+
+    @ColumnInfo(name = "tab_parent_id")
+    private String parentId;
+
+    @ColumnInfo(name = "tab_title")
+    private String title;
+
+    @ColumnInfo(name = "tab_url")
+    private String url;
+
+    /**
+     * Thumbnail uri for tab preview.
+     */
+    @ColumnInfo(name = "tab_thumbnail_uri")
+    private String thumbnailUri;
```

Comment:
Just to make sure: We didn't have the logic of serializing these URIs yet? (Or there are some Room magic here?)